proto/dydxprotocol/vault/share.proto

```diff
@@ -6,7 +6,7 @@ import "cosmos_proto/cosmos.proto";
 
 option go_package = "github.com/dydxprotocol/v4-chain/protocol/x/vault/types";
 
-// NumShares represents the number of shares in a vault.
```

Comment:
nit: Note that these are shares for the megavault?

Reply:
prefer to keep `NumShares` as just a generic concept of shares